packages/neuron-wallet/src/block-sync-renderer/sync/queue.ts

```diff
@@ -1,251 +1,208 @@
-import { LiveCellPersistor } from 'services/tx/livecell-persistor'
 import AssetAccountInfo from 'models/asset-account-info'
 import AssetAccountService from 'services/asset-account-service'
+import { Address as AddressInterface } from 'database/address/address-dao'
+import AddressParser from 'models/address-parser'
+import MultiSign from 'models/multi-sign'
+import IndexerConnector from './indexer-connector'
+import IndexerCacheService from './indexer-cache-service'
 
 export default class Queue {
   private lockHashes: string[]
+  private url: string
+  private addresses: AddressInterface[]
   private rpcService: RpcService
-
+  private indexerConnector: IndexerConnector | undefined
+  private checkAndSaveQueue: AsyncQueue<{transactions: Transaction[]}> | undefined
   private currentBlockNumber = BigInt(0)
-  private endBlockNumber = BigInt(0)
-  private rangeForCheck: RangeForCheck
-
-  private fetchSize: number = 4
-
-  private tipNumberSubject: BehaviorSubject<string>
-  private tipNumberListener: Subscription | undefined
-
-  private stopped: boolean = false
-  private inProcess: boolean = false
-
-  private yieldTime = 1
 
   private multiSignBlake160s: string[]
-
-  private assetAccountInfo: AssetAccountInfo | undefined
-
   private anyoneCanPayLockHashes: string[]
+  private assetAccountInfo: AssetAccountInfo
 
-  constructor(url: string, lockHashes: string[], anyoneCanPayLockHashes: string[], multiSignBlake160s: string[], startBlockNumber: bigint) {
-    this.lockHashes = lockHashes
-    this.currentBlockNumber = startBlockNumber
+  constructor(url: string, addresses: AddressInterface[]) {
+    this.url = url
+    this.addresses = addresses
     this.rpcService = new RpcService(url)
-    this.rangeForCheck = new RangeForCheck(url)
-    this.tipNumberSubject = NodeService.getInstance().tipNumberSubject
-    this.multiSignBlake160s = multiSignBlake160s
-    this.anyoneCanPayLockHashes = anyoneCanPayLockHashes
-
-    try {
-      this.assetAccountInfo = new AssetAccountInfo()
-    } catch(e) {
-      logger.info(`instance AssetAccountInfo error: ${e}`)
-    }
+    this.assetAccountInfo = new AssetAccountInfo()
+
+    this.lockHashes = AddressParser.batchToLockHash(
+      this.addresses.map(meta => meta.address)
+    )
+
+    const multiSign = new MultiSign()
+    const blake160s = this.addresses.map(meta => meta.blake160)
+    this.multiSignBlake160s = blake160s.map(blake160 => multiSign.hash(blake160))
+    this.anyoneCanPayLockHashes = blake160s.map(
+      b => this.assetAccountInfo.generateAnyoneCanPayScript(b).computeHash()
+    )
   }
 
-  public start = async () => {
-    await this.expandToTip(await this.rpcService.getTipBlockNumber())
-
-    this.tipNumberListener = this.tipNumberSubject.subscribe(async num => {
-      if (num) {
-        await this.expandToTip(num)
+  public async start() {
+    const indexedDataPath = path.resolve(
+      env.fileBasePath,
+      './indexer_data',
```

Comment:
Don't use magic string

Reply:
Could you please clarify a bit on what you mean by `magic string` here?

Reply:
A plain string without any indicators. In this case, the `indexer_data` seems to be a dir keeping databases.

Reply:
It is the folder path for holding the data built from lumos indexer? Any suggestions to better name this one from your view?

Reply:
Does `data built from lumos indexer` mean the `rocksdb` files? The `indexer_data` could be named as `LUMOS_INDEXER_DB_FOLDER` if it is.